README.md
# Regular Notifications

[![apm](https://img.shields.io/apm/v/regular-notifications.svg?style=flat-square)](https://atom.io/packages/regular-notifications)
[![apm](https://img.shields.io/apm/dm/regular-notifications.svg?style=flat-square)](https://atom.io/packages/regular-notifications)
[![apm](https://img.shields.io/apm/l/regular-notifications.svg?style=flat-square)](https://atom.io/packages/regular-notifications)

Display regular notifications at regular interval.  

![Starting Regular Notifications](https://raw.githubusercontent.com/iamsenthilprabu/regular-notifications/master/screenshots/regular-notifications.gif)

## Important

I renamed this package form 'custom-notifications' to 'regular-notifications'. If you have any issues with the package, kindly reinstall the package to make it work properly.

## Installation

Go to `Settings > Install` and then search for `regular-notifications` under `Packages`.

## Configuration

You can configure the notifications in regular-notifications -> settings.

![Configuration Screen](https://raw.githubusercontent.com/iamsenthilprabu/regular-notifications/master/screenshots/configuration.png)

You can customize the following:
 * Duration In Seconds: **Interval duration for the notification.**
 * Notification Text: **Notification Text.**

## Toogle (On / Off) the regular notification

 After Installing and configuring the package, you need to Turn ON the notification.

 For both Turn On and Turn Off the notification, you can do any of the following.

  * Command Palette: **Regular Notifications: Toggle**
  * Context Menu: **Right Click in Atom Editor -> Toggle Regular Notifications**
  * Keyboard Shortcut: **ctrl-alt-c**

When you Toogle regular Notifications, You will get a confirmation notification.

![Starting Regular Notifications](https://raw.githubusercontent.com/iamsenthilprabu/regular-notifications/master/screenshots/starting_regular_notification.png)

Then, you will get your regular notifications at regular interval.

![Regular Notifications](https://raw.githubusercontent.com/iamsenthilprabu/regular-notifications/master/screenshots/notification.png)

You can turn off the regular Notifications by using the same Toogle options.  
(Command Palette /Context Menu / Keyboard Shortcut)

![Stopping Regular Notifications](https://raw.githubusercontent.com/iamsenthilprabu/regular-notifications/master/screenshots/stoping_regular_notification.png)

### Contribute

Pull requests are welcome.  
If you like to add support for a language you work in, feel free to contribute and open a pull request.  
In lieu of a formal style guide, take care to maintain the existing coding style.

### License

Copyright (c) 2017 Senthilprabu Ponnusamy


CHANGELOG.md
## 0.1.0 - First Release
* Every feature added

## 0.1.1 - Second Release
* Package Renamed

## 0.5.0
* Package Renamed and issues fixed.


lib/custom-notifications.js
'use babel';

import { CompositeDisposable } from 'atom';

export default {
  timerId: null,
  timerFlag: false,
  subscriptions: null,
  remainingDurationInSeconds: 0,

  config: {
    customDurationInSeconds: {
      title: "Custom Duration In Seconds",
      type: 'integer',
      default: 600,
    },
    customNotificationText: {
      title: "Custom Notification Text",
      type: "string",
      default: "Drink Water / Commit your changes"
    }
  },

  activate(state) {
    this.subscriptions = new CompositeDisposable();
    this.subscriptions.add(atom.commands.add('atom-workspace', {
      'custom-notifications:toggle': () => this.toggle()
    }));
  },

  deactivate() {
    this.subscriptions.dispose();
  },

  toggle() {
    if (this.timerFlag) {
      this.stopTimer();
      this.timerFlag = false;
    } else {
      this.startTimer();
      this.timerFlag = true;
    }
  },

  startTimer() {
    atom.notifications.addSuccess('Custom notifications enabled');

    this.remainingDurationInSeconds = atom.config.get('custom-notifications.customDurationInSeconds');
    var timer_in_seconds = this.remainingDurationInSeconds % 60;
    var timer_in_minutes = (this.remainingDurationInSeconds - timer_in_seconds) / 60;
    var formated_timer = (timer_in_minutes > 0 ? timer_in_minutes + 'mins ' : '');
    if (timer_in_seconds) {
      formated_timer = formated_timer + timer_in_seconds + 'seconds';
    }
    atom.notifications.addInfo('You will receive your custom notification every: <br/>' + formated_timer);

    this.timerId = window.setInterval(
      (function(self) {
        return function() {
          self.processTimer();
        };
      })(this), 1000);
    },

    processTimer() {
      if (this.remainingDurationInSeconds <= 0) {
        atom.notifications.addInfo(atom.config.get('custom-notifications.customNotificationText'));
        this.remainingDurationInSeconds = atom.config.get('custom-notifications.customDurationInSeconds');
      } else {
        this.remainingDurationInSeconds--;
      }
    },

    stopTimer() {
      atom.notifications.addError('Custom notifications disabled.');
      window.clearInterval(this.timerId);
    },
  };


lib/regular-notifications.js
'use babel';

import { CompositeDisposable } from 'atom';

export default {
  timerId: null,
  timerFlag: false,
  subscriptions: null,
  remainingDurationInSeconds: 0,

  config: {
    DurationInSeconds: {
      title: "Duration In Seconds",
      type: 'integer',
      default: 600,
    },
    NotificationText: {
      title: "Notification Text",
      type: "string",
      default: "Drink Water / Commit your changes"
    }
  },

  activate(state) {
    this.subscriptions = new CompositeDisposable();
    this.subscriptions.add(atom.commands.add('atom-workspace', {
      'regular-notifications:toggle': () => this.toggle()
    }));
  },

  deactivate() {
    this.subscriptions.dispose();
  },

  toggle() {
    if (this.timerFlag) {
      this.stopTimer();
      this.timerFlag = false;
    } else {
      this.startTimer();
      this.timerFlag = true;
    }
  },

  startTimer() {
    atom.notifications.addSuccess('Regular notifications enabled');

    this.remainingDurationInSeconds = atom.config.get('regular-notifications.DurationInSeconds');
    var timer_in_seconds = this.remainingDurationInSeconds % 60;
    var timer_in_minutes = (this.remainingDurationInSeconds - timer_in_seconds) / 60;
    var formated_timer = (timer_in_minutes > 0 ? timer_in_minutes + 'mins ' : '');
    if (timer_in_seconds) {
      formated_timer = formated_timer + timer_in_seconds + 'seconds';
    }
    atom.notifications.addInfo('You will receive your regular notifications every: <br/>' + formated_timer);

    this.timerId = window.setInterval(
      (function(self) {
        return function() {
          self.processTimer();
        };
      })(this), 1000);
    },

    processTimer() {
      if (this.remainingDurationInSeconds <= 0) {
        atom.notifications.addInfo(atom.config.get('regular-notifications.NotificationText'));
        this.remainingDurationInSeconds = atom.config.get('regular-notifications.DurationInSeconds');
      } else {
        this.remainingDurationInSeconds--;
      }
    },

    stopTimer() {
      atom.notifications.addError('Regular notifications disabled.');
      window.clearInterval(this.timerId);
    },
  };
